ocf_datapipes/load/nwp/nwp.py

```diff
@@ -53,9 +123,37 @@ def __init__(
         else:
             raise ValueError(f"Unknown provider: {provider}")
 
-    def __iter__(self) -> Union[xr.DataArray, xr.Dataset]:
+    def __iter__(self) -> Union[xr.DataArray, xr.Dataset]:  # type: ignore
         """Opens the NWP data"""
         logger.debug("Opening NWP data: %s", self.zarr_path)
         nwp = self.open_nwp(self.zarr_path)
+        if self.check_for_zeros:
+            self.check_if_zeros(nwp)
+        if self.check_physical_limits:
+            self.check_if_physical_limits(nwp)
         while True:
             yield nwp
+
+    def check_if_zeros(self, nwp: Union[xr.DataArray, xr.Dataset]):
+        """Checks if the NWP data contains zeros"""
+        if isinstance(nwp, xr.DataArray):
+            if (nwp.values == 0).any():
+                raise ValueError("NWP DataArray contains zeros")
+        if isinstance(nwp, xr.Dataset):
+            for var in nwp:
+                if (nwp[var].values == 0).any():
+                    raise ValueError(f"NWP Dataset variable{var} contains zeros")
+
+    def check_if_physical_limits(self, nwp: Union[xr.DataArray, xr.Dataset]):
+        """Checks if the NWP data is within physical limits"""
+        if isinstance(nwp, xr.DataArray):
+            var_name = nwp.name
+            if var_name in self.limits:
+                lower, upper = self.limits[var_name]
+                if (nwp < lower).any() or (nwp > upper).any():
+                    raise ValueError(f"NWP data {var_name} is outside physical limits")
```

Comment:
can you add what the upper and lower limit are, and perhaps print the max and min in the error statement, same below?